### PR TITLE
Convert all references and instances of PoolVector to Vector

### PR DESCRIPTION
**Pull request by reduz**
Makes no sense to keep PoolVector in a world of 64 bits computers.

### CODE REVIEW

**Comment by fire** [comment]
Since someone asked. Here's the design proposal for converting PoolVector to Vector. 

https://docs.google.com/document/d/1HtFfENFdnmzB1XhviIqVoeGQbIB1x30vAIFtjTFW18o/edit

**Comment by aaronfranke** [comment]
I assume `PoolWhateverArray` would need to be renamed to `PackedWhateverArray` in user code and projects, but does it behave the same, or is the behavior different?

Is there any reason to keep using `Vector<String>` internally when we could just use the `PackedStringArray` semantic? It seems that right now they are used interchangeably. If the concern is with too many things depending on `variant.h` since it includes the typdefs, we could potentially move the typedefs to a different header file, and `PackedByteArray`/`PackedIntArray`/`PackedRealArray` don't need anything other than `vector.h` and `math_defs.h`.

**Comment by akien-mga** [comment]
I rebased to squash the commits and amended to fix build for Mono, Windows and macOS.

There are still a few fixes needed for Android, iOS and JavaScript, but since they can't build right now due to the lack of rendering backend, I prefer not to do the changes blindly.

**Comment by reduz** [comment]
@aaronfranke no the names are only in the context of Variant and used for exposing API and serialziing. They are called packed because regular arays ini Variant are just arrays of variants.

Some contributors used them in their code for unrelated stuff but its not really required.

**Comment by akien-mga** [comment]
Thanks!

File: core/bind/core_bind.cpp [hunk 1]
```diff
@@ -646,11 +646,6 @@ uint64_t _OS::get_static_memory_peak_usage() const {
 	return OS::get_singleton()->get_static_memory_peak_usage();
 }
 
-uint64_t _OS::get_dynamic_memory_usage() const {
```
Comment (natrim) [inline core/bind/core_bind.cpp]: was this removed by mistake?

Comment (akien-mga) [reply core/bind/core_bind.cpp]: I assume yes, as the `MemoryPool` this referred to was also removed: https://github.com/godotengine/godot/pull/36311/files/3205a92ad872f918c8322cdcd1434c231a1fd251#diff-ab1978b28d0ec282d1df67c7903b6191L419-L422

**Comment by agar3s** [comment]
Should be this section in the docs updated?

> For dynamic memory, the PoolVector<> template is provided. PoolVector is a standard vector class, and is very similar to vector in the C++ standard library. To create a PoolVector buffer, use this:

https://docs.godotengine.org/en/latest/development/cpp/core_types.html#allocating-memory

**Comment by clayjohn** [comment]
@agar3s Yes. Please make an issue in the docs repo so other contributors can keep track. Or make a PR yourself if you are up to it. :)

**Comment by agar3s** [comment]
@clayjohn there is an issue already: https://github.com/godotengine/godot-docs/issues/3192

**Pull request merged**